tests/executables_src/testStatusMonitor.cc

```diff
@@ -0,0 +1,961 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#include <future>
+
+#define BOOST_TEST_MODULE testExceptionHandling
+
+#include "Application.h"
+#include "ControlSystemModule.h"
+#include "ScalarAccessor.h"
+#include "StatusMonitor.h"
+#include "TestFacility.h"
+
+#include <boost/mpl/list.hpp>
+#include <boost/test/included/unit_test.hpp>
+
+using namespace boost::unit_test_framework;
+namespace ctk = ChimeraTK;
+
+/* dummy application - for new StatusMonitor interface */
+template<typename T>
+struct TestApplication : public ctk::Application {
+  TestApplication() : Application("testSuite") {}
+  ~TestApplication() override { shutdown(); }
+
+  void defineConnections() override {
+    findTag(".*").connectTo(cs); // publish everything to CS
+    findTag("MON_PARAMS")
+        .connectTo(cs["MonitorParameters"]); // cable the parameters in addition (checking that tags are set correctly)
+    findTag("MON_OUTPUT")
+        .connectTo(cs["MonitorOutput"]); // cable the parameters in addition (checking that tags are set correctly)
+  }
+  ctk::ControlSystemModule cs;
+  T monitor{this, "/input/path", "/output/path", "/parameters", "Now this is a nice monitor...",
+      ctk::TAGS{"MON_OUTPUT"}, ctk::TAGS{"MON_PARAMS"}};
+};
+
+/*********************************************************************************************************************/
+
+BOOST_AUTO_TEST_CASE(testMonitorDataValidityPropagation) {
```

Comment:
Should be called `testMaxMonitorDataValidityPropagation`, shouldn't it?

Reply:
Not really, it's testing general data validity by only using MaxMonitor, but this suppose to be generic. Data validity is in base class, so its used by all monitors, but we cannot instanciate the base class as it does not implement any real functionality. I tried to explain it in the comment in beginning of the test.

Reply:
Ah yes of course, sorry. Maybe name the test then simply `testDataValidityPropagation`, as that looks less like a typo, and place the comment above the test not inside... :-)